src/Nethermind/Nethermind.Core/Collections/LockableConcurrentDictionary.cs

```diff
@@ -22,7 +22,7 @@ public static class ConcurrentDictionaryLock<TKey, TValue> where TKey : notnull
     /// <summary>
     /// Delegate that is equivalent of <see cref="ConcurrentDictionary{TKey,TValue}.ReleaseLocks"/>
     /// </summary>
-    private delegate void ReleaseLocks(ConcurrentDictionary<TKey, TValue> dictionary, int fromInclusive, int toExclusive);
+    private delegate void ReleaseLocks(ConcurrentDictionary<TKey, TValue> dictionary, int locksAcquired);
```

Comment:
This class is an illustrious example of why relying on private APIs is not a good idea. The signature of the method [`ReleaseLocks`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs#L2108) has been changed in .NET 8 breaking our current implementation.

@LukaszRozmej Do you have any idea on how to make this future-proof?

Reply:
I don't think we can make it future-proof, the test is there just for this and it worked. So I think we are good.